config_converter/config_mapper/config_mapper.py

```diff
@@ -30,15 +30,51 @@
 _SUPPORTED_PLUGINS = ['in_tail']
 
 
+def initialize_stats(d: config_pb2.Directive) -> dict:
+    """Initializes the stats dict to print out."""
+    stats = {
+        'attributes_num': get_num_attrs(d),
+        'attributes_recognized': 0,
+        'attributes_unrecognized': 0,
+        'attributes_skipped': 0,
+        'entities_num': get_num_ents(d),
+        'entities_skipped': 0,
+        'entities_unrecognized': 0,
+        'entities_recognized_success': 0,
+        'entities_recognized_partial': 0,
+        'entities_recognized_failure': 0
+    }
+    return stats
+
+
+def get_num_attrs(d: config_pb2.Directive) -> int:
+    """Takes a directive, returns total no of attributes."""
+    num_attrs = len(list(d.params))
+    for nd in d.directives:
+        num_attrs += get_num_attrs(nd)
+    return num_attrs
+
+
+def get_num_ents(d: config_pb2.Directive) -> int:
+    """Takes a directive, returns total no of directives."""
```

Comment:
```suggestion
    """Takes a directive, returns total number of directives rooted from this directives (not including itself)."""
```

as itself is a directive, maybe call `get_num_sub_entities`?

Reply:
Good suggestion - I can change that!